lib/http-bridge/src/Cardano/Wallet/HttpBridge/Transaction.hs

```diff
@@ -65,8 +69,6 @@ newTransactionLayer = TransactionLayer
     { mkStdTx = \keyFrom inps outs -> do
         let ins = (fmap fst inps)
         let tx = Tx ins outs
-        when (any (\ (TxOut _ c) -> c == Coin 0) outs)
-            $ Left ErrInvalidTx
         let txSigData = txId @(HttpBridge n) tx
```

Comment:
So, `validateSelection` is now called from the upper layer in `Cardano.Wallet`, though it should really be called from within `mkStdTx`. Otherwise, it becomes the caller responsibility to know that they have to validate the selection before calling `mkStdTx` which may cause hard to catch errors later on. Since we don't guarantee at the type-level that a selection is valid, then, we have to put this check / invariant in the `mkStdTx` function where it actually matters for the selection to be valid!

Reply:
now `validateSelection` used inside `mkStdTx`